src/lib/utils/loadstor.h

```diff
@@ -115,169 +166,87 @@ inline constexpr T load_le(const uint8_t in[], size_t off) {
-#endif
-}
-
-/**
-* Load a little-endian uint16_t
-* @param in a pointer to some bytes
-* @param off an offset into the array
-* @return off'th uint16_t of in, as a little-endian value
-*/
-template <>
-inline constexpr uint16_t load_le<uint16_t>(const uint8_t in[], size_t off) {
-   in += off * sizeof(uint16_t);
-
-#if defined(BOTAN_TARGET_CPU_IS_BIG_ENDIAN)
-   return reverse_bytes(typecast_copy<uint16_t>(in));
-#elif defined(BOTAN_TARGET_CPU_IS_LITTLE_ENDIAN)
-   return typecast_copy<uint16_t>(in);
-#else
-   return make_uint16(in[1], in[0]);
-#endif
-}
-
-/**
-* Load a big-endian uint32_t
-* @param in a pointer to some bytes
-* @param off an offset into the array
-* @return off'th uint32_t of in, as a big-endian value
-*/
-template <>
-inline constexpr uint32_t load_be<uint32_t>(const uint8_t in[], size_t off) {
-   in += off * sizeof(uint32_t);
-
-#if defined(BOTAN_TARGET_CPU_IS_BIG_ENDIAN)
-   return typecast_copy<uint32_t>(in);
-#elif defined(BOTAN_TARGET_CPU_IS_LITTLE_ENDIAN)
-   return reverse_bytes(typecast_copy<uint32_t>(in));
-#else
-   return make_uint32(in[0], in[1], in[2], in[3]);
-#endif
+template <concepts::unsigned_integral T>
+inline constexpr T load_be(const uint8_t in[], size_t off) {
+   // asserts that *in points to the correct amount of memory
+   return load_be<T>(std::span<const uint8_t, sizeof(T)>(in + off * sizeof(T), sizeof(T)));
 }
 
 /**
-* Load a little-endian uint32_t
+* Load a little-endian unsigned integer
 * @param in a pointer to some bytes
 * @param off an offset into the array
-* @return off'th uint32_t of in, as a little-endian value
+* @return off'th unsigned integer of in, as a little-endian value
 */
-template <>
-inline constexpr uint32_t load_le<uint32_t>(const uint8_t in[], size_t off) {
-   in += off * sizeof(uint32_t);
-
-#if defined(BOTAN_TARGET_CPU_IS_BIG_ENDIAN)
-   return reverse_bytes(typecast_copy<uint32_t>(in));
-#elif defined(BOTAN_TARGET_CPU_IS_LITTLE_ENDIAN)
-   return typecast_copy<uint32_t>(in);
-#else
-   return make_uint32(in[3], in[2], in[1], in[0]);
-#endif
+template <concepts::unsigned_integral T>
+inline constexpr T load_le(const uint8_t in[], size_t off) {
+   // asserts that *in points to the correct amount of memory
+   return load_le<T>(std::span<const uint8_t, sizeof(T)>(in + off * sizeof(T), sizeof(T)));
 }
 
 /**
-* Load a big-endian uint64_t
-* @param in a pointer to some bytes
-* @param off an offset into the array
-* @return off'th uint64_t of in, as a big-endian value
+* Load many big-endian unsigned integers
+* @param in   a fixed-length span to some bytes
+* @param outs a arbitrary-length parameter list of unsigned integers to be loaded
 */
-template <>
-inline constexpr uint64_t load_be<uint64_t>(const uint8_t in[], size_t off) {
-   in += off * sizeof(uint64_t);
+template <ranges::contiguous_range<uint8_t> InR, concepts::unsigned_integral... Ts>
+   requires all_same_v<Ts...>
+inline constexpr void load_be(InR&& in, Ts&... outs) {
```

Comment:
Do we need tests for multiple _out_ params for load_le/be (or multiple _in_ params for store_le/be)?